dev/PushNotifications/PushNotificationUtility.h

```diff
@@ -127,9 +130,18 @@ namespace winrt::Microsoft::Windows::PushNotifications::Helpers
     }
     CATCH_RETURN()
 
+    inline void ClearRoActivateInstanceCache()
+    {
+        // Activating a nonexistent class clears the cache enables the proxyStub to be found for elevated scenarios
+        auto activatableClass{ wil::make_unique_string_nothrow<wil::unique_hstring>(L"FakeClass.ClassName") };
+        winrt::com_ptr<::IInspectable> ptr{};
+
+        RoActivateInstance(activatableClass.get(), ptr.put());
```

Comment:
No check of the return code?

Reply:
What check should be performed? This activation is supposed to fail. You _could_ validate that it fails with the expected REGDB_E_CLASSNOTREG, but that's probably over-validating. If it somehow succeeds (e.g. if someone actually implements a class named "FakeClass.ClassName" within the package graph; or if somebody detours RoActivateInstance to always succeed), I guess you _could_ crash, but barring that you would just want to clean up the activated object and move on, which is done through RAII.

From a style perspective it's probably worth commenting and maybe casting the return to void, just to make it clear that it's being ignored deliberately. But functionally this should be fine.

Reply:
Adding comment and cast to explain impl.

Reply:
Actually, I'm going to check for the REGDB_E_CLASSNOTREG. It feels weird to not verify the return value.

Reply:
It's ultimately your choice. Error codes are a bit of a gray area in terms of "is this an implementation detail, or a stable part of the API's contract?" My general view is that if an API doesn't specifically document that it returns a specific error code for a given failure mode, then taking a dependency on that detail risks getting broken if the API's behavior changes in the future.

But with that said, there's enough precedent for checking for REGDB_E_CLASSNOTREG from COM/WinRT activations that it's definitely de facto part of the contract of those APIs. I'd say it's extremely unlikely we'd ever be able to change it, due to compat risk. So in practice, validating that specific error code is unlikely to be harmful.